lib/ansible/modules/cloud/cloudstack/cs_instance_password_reset.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# (c) 2018, Gregor Riepl <onitake@gmail.com>
+# based on cs_sshkeypair (c) 2015, René Moser <mail@renemoser.net>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible. If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: cs_instance_password_reset
+short_description: Allows resetting VM the default passwords on Apache CloudStack based clouds.
+description:
+    - Resets the default user account's password on an instance.
+    - Requires cloud-init to be installed in the virtual machine.
+    - The passwordenabled flag must be set on the template associated with the VM.
+version_added: '2.8'
+author: "Gregor Riepl (@onitake)"
+options:
+  vm:
+    description:
+      - Name of the virtual machine to reset the password on.
+    required: true
+  domain:
+    description:
+      - Name of the domain the virtual machine belongs to.
+  account:
+    description:
+      - Account the virtual machine belongs to.
+  project:
+    description:
+      - Name of the project the virtual machine belongs to.
+  zone:
+    description:
+      - Name of the zone in which the instance is deployed.
+      - If not set, the default zone is used.
+  poll_async:
+    description:
+      - Poll async jobs until job has finished.
+    default: yes
+    type: bool
+extends_documentation_fragment: cloudstack
+'''
+
+EXAMPLES = '''
```

Comment:
Shouldn't the vm be stopped before we can reset it's password?

Reply:
It should, yes. Fixed.